lib/websocket.js

```diff
@@ -27,26 +27,124 @@ function init (env, ctx, server) {
       'browser client gzip': false
     });
   }
+  
+  function verifyAuthorization(secret) {
+    var read, write, write_treatment;
+    
+    // read now true by default
+    read = true;
+    write = (secret === env.api_secret);
+    write_treatment = !env.treatments_auth || (secret === env.api_secret);
+    
+    return {
+      read: read
+      , write: write
+      , write_treatment: write_treatment
+    };
+  }
 
   function emitData (delta) {
     if (lastData.cals) {
       console.log('running websocket.emitData', ctx.data.lastUpdated);
-      io.emit('dataUpdate', delta);
+      io.to('DataReceivers').emit('dataUpdate', delta);
     }
   }
 
   function listeners ( ) {
     io.sockets.on('connection', function (socket) {
-      // send all data upon new connection
-      socket.emit('dataUpdate',lastData);
+      var socketAuthorization = null;
+      var clientType = null;
+      
+      console.log('Connection from client ID: ', socket.client.id, ' IP: ', socket.request.connection.remoteAddress);
+
       io.emit('clients', ++watchers);
       socket.on('ack', function(alarmType, silenceTime) {
         //TODO: Announcement hack a1/a2
         var level = alarmType2Level[alarmType] || alarmType;
         ctx.notifications.ack(level, silenceTime, true);
       });
+
       socket.on('disconnect', function () {
         io.emit('clients', --watchers);
+        console.log('Disconnected client ID: ',socket.client.id);
+      });
+
+
+      // dbUpdate message  
+      //  {
+      //    collection: treatments
+      //    _id: 'some mongo record id'
+      //    data: {
+      //      field_1: new_value,
+      //      field_2: another_value
+      //    }
+      //  }
+      socket.on('dbUpdate', function dbUpdate(data, callback) {
+        var ObjectID = require('mongodb').ObjectID;
+       
+        // TODO: this would be better to have somehow integrated/improved
+        var supportedCollections = { 
+        'treatments' : env.treatments_collection, 
+        'entries': env.mongo_collection, 
+        'devicestatus': env.devicestatus_collection, 
+        'profile': env.profile_collection, 
+        'food': env.food_collection
+        };
+
+        var collection = supportedCollections[data.collection];
+        
+        if (!collection) {
+          console.log('WS dbUpdate call: ', 'Wrong collection', data);
+          return { result: 'Wrong collection' };
+        }
+
+        if (!socketAuthorization) {
+          console.log('WS dbUpdate call: ', 'Not authorized', data);
+          return { result: 'Not authorized' };
+        }
+
+        if (data.collection == 'treatments') {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1644868632/issues/source?bid=2452379&fileBranchId=3019946#l106)